model/src/pyrenew/latent/infection_seeding_process.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+import numpyro as npro
+import numpyro.distributions as dist
+from pyrenew.latent.infection_seeding_method import InfectionSeedMethod
+from pyrenew.metaclass import RandomVariable
+
+
+class InfectionSeedingProcess(RandomVariable):
+    r"""Infection Seeding Process"""
+
+    def __init__(
+        self,
+        I0_dist: dist.Distribution,
+        infection_seed_method: InfectionSeedMethod,
+    ) -> None:
+        InfectionSeedingProcess.validate(I0_dist, infection_seed_method)
+
+        self.I0_dist = I0_dist
+        self.infection_seed_method = infection_seed_method
+
+        return None
+
+    @staticmethod
+    def validate(
+        I0_dist: dist.Distribution,
+        infection_seed_method: InfectionSeedMethod,
+    ) -> None:
+        assert isinstance(I0_dist, dist.Distribution)
+        assert isinstance(infection_seed_method, InfectionSeedMethod)
+
+    def sample(self) -> tuple:
+        I0 = npro.sample("I0", self.I0_dist)
+
+        infection_seeding = self.infection_seed_method.seed_infections(I0)
+        I0_det = npro.deterministic("I0", infection_seeding)
```

Comment:
Copying from https://github.com/CDCgov/multisignal-epi-inference/blob/main/model/src/pyrenew/process/rtrandomwalk.py#L123

What is the point of declaring this `deterministic`? It confused me because `I0` is being sampled.

Reply:
`numpyro.deterministic` labels deterministic functions of other parameters for tracking/recording during MCMC. Similar to declaring a `transformed parameter` or deterministic `generated quantity` in Stan. As written we probably don't want this, as it gives the computed quantity the same tracing name ("`I0`") as the upstream directly-sampled quantity from which it is produced.